packages/astro/src/vite-plugin-astro-server/index.ts

```diff
@@ -195,7 +195,7 @@ async function handleRequest(
 	const url = new URL(origin + req.url?.replace(/(index)?\.html$/, ''));
 	const pathname = decodeURI(url.pathname);
 	const rootRelativeUrl = pathname.substring(devRoot.length - 1);
-	if (!buildingToSSR) {
+	if (!buildingToSSR && viteServer.config.mode === 'production') {
```

Comment:
This makes sense since this integration needs query params in `dev` for SSG, but won't this allow users to accidentally depend on search params in SSG mode?